test/utils.go

```diff
@@ -21,7 +23,13 @@ func RandPeerID() (peer.ID, error) {
 }
 
 func RandTestKeyPair(bits int) (ci.PrivKey, ci.PubKey, error) {
-	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+	seed := time.Now().UnixNano()
+
+	// workaround for low time resolution
+	seed += (1 << 32) * generatedPairs
+	generatedPairs++
```

Comment:
This will need to use an atomic. We have some packages (e.g., swarm) that run tests in parallel.

Reply:
done